### PR TITLE
fix remoteproc get wrong memory when memory region are continuous.

### PR DESCRIPTION
**Pull request by joshualin-petaio**

### CODE REVIEW

File: lib/remoteproc/remoteproc.c [hunk 1]
```diff
@@ -55,14 +55,14 @@ remoteproc_get_mem(struct remoteproc *rproc, const char *name,
 
 			pa_start = mem->pa;
 			pa_end = pa_start + mem->size;
-			if (pa >= pa_start && (pa + size) <= pa_end)
+			if (pa >= pa_start && (pa + size) < pa_end)
```
Comment (arnopo) [inline lib/remoteproc/remoteproc.c:58]: (pa + size) = pa_end is a valid use case if size is not equal to 0.
That why i suggested you in https://github.com/OpenAMP/open-amp/issues/335 to return an error if pa >= pa_end

**Review by arnopo — CHANGES_REQUESTED** [review]
Please rewrite the commit message (cf compliance CI fail).
A valid commit message contains:
- a text that give the rational of the patch
- a signed_of

File: lib/remoteproc/remoteproc.c (hunk 1 above)
Comment (joshualin-petaio) [reply lib/remoteproc/remoteproc.c:58]: Hi Arnopo,

Thank you for your explain.
The pa >= pa_end condition is really a best method.
I will modify it and send PR again.

Thanks,
Joshua

**Comment by joshualin-petaio** [comment]
Sorry about that I close this PR.
There has some issue about rewrite commit.
I will re-send the PR include issue #335 and #338.
Thank you.

**Pull request closed without merging**